### PR TITLE
[terra-menu] Add theme context

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
Add theme context to terra-menu

### Deployment Link

https://terra-framework-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-menu/src/MenuItem.module.scss [hunk 1]
```diff
@@ -1,3 +1,6 @@
+@import './clinical-lowlight-theme/Menu.module';
+@import './orion-fusion-theme/Menu.module';
```
Comment (emilyrohrbough) [inline packages/terra-menu/src/MenuItem.module.scss:2]: These should be imported from MenuItem.moudle.scss

File: packages/terra-menu/src/_MenuContent.jsx [hunk 2]
```diff
@@ -340,12 +341,14 @@ class MenuContent extends React.Component {
...
     );
+    const theme = this.context;
     const isSubMenu = this.props.index > 0;
     const contentClass = cx([
       'content',
       { submenu: isSubMenu },
       { 'hidden-page': this.props.isHidden },
       { fullscreen: isFullScreen },
+      theme.className,
     ]);
```
Comment (emilyrohrbough) [inline packages/terra-menu/src/_MenuContent.jsx:352]:
```suggestion
    const contentClass = cx(
      'content',
      { submenu: isSubMenu },
      { 'hidden-page': this.props.isHidden },
      { fullscreen: isFullScreen },
      theme.className,
    );
```

File: packages/terra-menu/src/clinical-lowlight-theme/Menu.module.scss [hunk 3]
```diff
@@ -0,0 +1,78 @@
+:local {
+  .clinical-lowlight-theme {
```
Comment (emilyrohrbough) [inline packages/terra-menu/src/clinical-lowlight-theme/Menu.module.scss:2]: Let's break the MenuItem theme vars into their own file.

File: packages/terra-menu/src/MenuItem.module.scss (hunk 1 above)
Comment (saket2403) [reply packages/terra-menu/src/MenuItem.module.scss]: Done cf26d44

File: packages/terra-menu/src/clinical-lowlight-theme/Menu.module.scss (hunk 3 above)
Comment (saket2403) [reply packages/terra-menu/src/clinical-lowlight-theme/Menu.module.scss:2]: Done cf26d44

**Comment by saket2403** [comment]
> Can we add jest test for this change?

Added b4e7d35

**Pull request merged**